### ISSUE TITLE
[MNG-7185] Describe explicit and recommended version for VersionRange.createFromVersionSpec()

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#8272]
**[Nils Breunese](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=breun)** opened **[MNG-7185](https://issues.apache.org/jira/browse/MNG-7185?redirect=false)** and commented

I would expect a version range for a single version to not contain any other versions, but it seems this is not the case, because this test fails on the second assertion:

```
@Test
void range_with_single_version_should_not_contain_other_version() {
        VersionRange singleVersionRange = VersionRange.createFromVersionSpec("1.0.0");

        assertThat(singleVersionRange.containsVersion(new DefaultArtifactVersion("1.0.0"))).isTrue();
        assertThat(singleVersionRange.containsVersion(new DefaultArtifactVersion("2.0.0"))).isFalse();
}
```

Is this a bug, or do I misinterpret what a single version range is? Does `maven-artifact` have a concept for a version range that only contains a single version?

---

**Affects:** 3.8.1

**Remote Links:**
- [GitHub Pull Request #487
](https://github.com/apache/maven/pull/487)

**Backported to:** [4.0.0-alpha-2](https://github.com/apache/maven/milestone/113?closed=1), [3.8.2](https://github.com/apache/maven/milestone/103?closed=1)

**Comment by jira-importer** [comment]
**[Nils Breunese](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=breun)** commented

Thanks for the explanation. I guess the conclusion is that there is no bug, but Javadoc for `VersionRange#createFromVersionSpec` doesn't show the syntax for a closed range for a single version.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [e29a6610865f674f6d9b856c03d156074d0d2df8](https://gitbox.apache.org/repos/asf?p=maven.git;a=commit;h=e29a6610865f674f6d9b856c03d156074d0d2df8) and [176b272f30c4fbd62013b4702ab28518c21628ac](https://gitbox.apache.org/repos/asf?p=maven.git;a=commit;h=176b272f30c4fbd62013b4702ab28518c21628ac) for `maven-3.8.x` branch.
